### PR TITLE
Resolves feature request : #541 Add Oracle compatibility

### PR DESCRIPTION
**Pull request by andye2004**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -54,6 +54,14 @@ Integrates with Spring Data, Spring Data REST and Apache Solr</description>
         <docs.output.dir>../target/generated-docs/refs/${env.BUILD_TYPE}/</docs.output.dir>
     </properties>
 
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-slf4j-impl</artifactId>
+            <version>2.11.2</version>
+        </dependency>
```
Comment (andye2004) [inline pom.xml:62]: Resolves slf4j static logger warnings.

**Comment by andye2004** [comment]
Hi @paulcwarren, firstly, apologies on two fronts.

I have been unintentionally spamming your PR Validation Actions queue, I didn't realise this would happen with an open PR as I'm not really familiar with github actions. Obviously, it's only the very last item in the queue titled 'Resolves feature request....' that will need to be processed, the others can be discarded. Also, my IDE discards certain whitespace characters based on a development profile used by my employers and this has modified some files, most notable in the maven.yml file. I have tried to revert as many of these changes as possible but some have sneaked through and I hope these don't cause you any concern.

Now, onto the changes.

I've included the addition of TestContainers for the Oracle tests and I've also added a little bit of documentation in the JPA section. I have not added myself as an `@author` as requested by the contributors agreement as I don't think anything I have committed warrants it. If you feel there is somewhere you think it should be added let me know via the comments.

On the issue of TestContainers, my builds fail on the branch due to S3 and Azure dependencies, whilst I have no experience with either, there are test containers that could maybe be used in place of the existing test dependencies that would make the tests less brittle in no paulcwarren CI environments. I'd be happy to help out in converting some of the existing test dependencies if it was something you wanted to look at.

Until next time.....

File: spring-content-rest/pom.xml [hunk 2]
```diff
@@ -220,12 +220,6 @@
             <version>4.1.1</version>
             <scope>test</scope>
         </dependency>
-        
-        <dependency>
-            <groupId>org.springframework.data</groupId>
-            <artifactId>spring-data-mongodb</artifactId>
-            <scope>test</scope>
-        </dependency>
```
Comment (andye2004) [inline spring-content-rest/pom.xml:228]: removed replicated dependency

**Comment by paulcwarren** [comment]
Thanks @andye2004 , looks good and no need to add any oracle specific code  👍 

I had to approve the build and it failed but it's unrelated to your changes.  I see several dependabot PRs also failing with the same issue.  Seems weird as literally nothing changed.  I'll merge as soon as I figure that out.

Thank you very much for the contribution.

**Comment by andye2004** [comment]
> Thank you very much for the contribution.

Pleased to help out. Just a quick heads-up, I spotted a bug in the version drop tables SQL which I pushed a fix for.

**Comment by paulcwarren** [comment]
Yeah, this passes locally so I will merge.

In the end I figured out that Github Actions doesn't share secrets with PRs from forked repo's (security hole) and this is what is causing the failures (and the dependabot ones too actually - original dependabot didn't work this way).  So, moving to TestContainers where we don't have to rely on real services seems like the way to go.

**Pull request merged**